src/core/gosling-track-model.ts

```diff
@@ -599,6 +600,7 @@ export class GoslingTrackModel {
                     }
                 } else if (IsChannelDeep(channel) && (channel.type === 'quantitative' || channel.type === 'genomic')) {
                     if (channel.domain === undefined && field) {
+                        // TOOD: do this for all fields
```

Comment:
`domain` should be the union of `startField`, `endField`, `startField2`, `endField2` if multiple fields are used.